server/routes/login.js

```diff
@@ -1,207 +1,33 @@
-
-      // Add Api health check
-      healthRes = await health.healthResult([
-        `${userAuthentication.ApiUrl}health`,
-        `${userAuthentication.ApiUrl}health/invision`,
-      ])
-      isApiUp = healthRes.status === 200
-      log.info(`loginIndex - health check called and the isAppUp = ${isApiUp} with status ${healthRes.status}`)
-      log.info(`${userAuthentication.ApiUrl}health - health check with status ${healthRes.status}`)
-
-      if (isApiUp === false) {
-        log.error(healthRes.appInfo)
-        res.render('pages/index', {
-          showUserNotSignedUpMessage: false,
-          authError: false,
-          apiUp: isApiUp,
-          csrfToken: res.locals.csrfToken,
-        })
-        return
-      }
-
-      const quantumAddresses = config.quantumAddresses.split(',')
-
-      if (config.twoFactorAuthOn === 'true' && ipRangeCheck(ipAddress, quantumAddresses) === false) {
-        if (userAuthenticationDetails === null || userAuthenticationDetails.length === 0) {
-          throw new Error(`Error : No Sms or Email address returned for QuantumId : ${req.user.username}`)
-        }
-
-        // eslint-disable-next-line no-shadow
-        const userAuthentication = userAuthenticationDetails[0]
-
-        if (
-          (userAuthentication.EmailAddress === null || userAuthentication.EmailAddress === '') &&
-          (userAuthentication.Sms === null || userAuthentication.Sms === '')
-        ) {
-          throw new Error(`Error : Sms or Email address null or empty for QuantumId : ${req.user.username}`)
-        }
-
-        const emailEnabled = userAuthentication.UseEmailAddress
-        const smsEnabled = userAuthentication.UseSms
-
-        if (!emailEnabled && !smsEnabled) {
-          throw new Error(`Error : Sms or Email address both set to false for QuantumId : ${req.user.username}`)
-        }
-
-        const twofactorCode = utilities.get2faCode()
-
-        await userAuthenticationService.updateTwoFactorAuthenticationHash(
-          req.user.username,
-          utilities.createTwoFactorAuthenticationHash(twofactorCode.toString()),
-        )
-
-        if (smsEnabled) {
-          // For SMS
-          await notify
-            .sendSms(notifySmsTemplate, userAuthentication.Sms || '', {
-              personalisation: { '2fa_code': twofactorCode },
-            })
-            .catch((err) => {
-              throw new Error(err)
-            })
-        }
-
-        if (emailEnabled) {
-          // For email
-          await notify
-            .sendEmail(notifyEmailTemplate, userAuthentication.EmailAddress || '', {
-              personalisation: { '2fa_code': twofactorCode },
-            })
-            .catch((err) => {
-              throw new Error(err)
-            })
-        }
-
+      const userAuthenticationDetails = userAuthenticationService.getUserAuthenticationDetails(req.user.username)
+
+      if (
+        userHasRole(req.user.token, 'ROLE_MFA') === false &&
```

Comment:
Non blocking - usually I would pull out this type of boolean logic into a variable for readability. Is it possible to do this or add a comment? That said none of the other code is written like this so may just be a personal preference of mine in which case ignore :)